### PR TITLE
523-canary-add-edit

### LINKED ISSUE #523: Canary - Add / Edit
- From the health page, provide an option to add a  new health - This mapts to Settings -> Health -> Add, 
- For existing checks created through this mechanism provide an edit/update options
- For checks created via other mechanisms (i.e whith a non null source, disable the edit functionaly, with an explanation of why and a link to click on for the url)

### PR DESCRIPTION
**Pull request by mainawycliffe**
This PR is built on top of #450

Closes #523

### CODE REVIEW

**Comment by moshloop** [comment]
@mainawycliffe are there refactoring / typing in this single commit ? those need to go into their own commit

**Comment by moshloop** [comment]
Can we add a tab called spec, that shows the spec?

**Comment by moshloop** [comment]
We also seem to have introduced  `"-"` vs `-`

**Comment by mainawycliffe** [comment]
> @mainawycliffe are there refactoring / typing in this single commit ? those need to go into their own commit

There are a few but minimal, let me pull them out.

**Comment by mainawycliffe** [comment]
> We also seem to have introduced `"-"` vs `-`

Let me fix this one

**Comment by mainawycliffe** [comment]
> We also seem to have introduced `"-"` vs `-`

Let me fix this one

**Comment by mainawycliffe** [comment]
> Can we add a tab called spec, that shows the spec?

@moshloop Yeah, just wondering whether I should do this as part of this PR or #610?

**Comment by mainawycliffe** [comment]
@moshloop I have done the spec tab as part #610 as it makes more sense there, and there we a lot of refactoring in that PR that was needed to wire everything for the spec tab.

File: src/api/services/topology.ts [hunk 1]
```diff
@@ -65,5 +70,33 @@ export const getTopologyComponentLabels = () => {
...
+  updated_at: string;
+  schedule: string;
+  deleted_at: string;
+};
+
+export const getComponentTemplate = async (id: string) => {
+  const res = await IncidentCommander.get<ComponentTemplateItem[]>(
+    `/templates?id=eq.${id}`
+  );
+  return res.data[0];
```
Comment (Prashanth-Nelli) [inline src/api/services/topology.ts:94]: please handle null value cases

File: src/components/Canary/CanaryPopup/CheckTitle.tsx [hunk 2]
```diff
@@ -4,7 +4,18 @@ import { usePrevious } from "../../../utils/hooks";
 import { Badge } from "../../Badge";
 import { Icon } from "../../Icon";
 
-export function CheckTitle({ check, className, size = "large", ...rest }) {
+type Props = {
+  check: any;
+  className?: string;
```
Comment (Prashanth-Nelli) [inline src/components/Canary/CanaryPopup/CheckTitle.tsx:9]: className prop is already there in the type `React.HTMLAttributes<HTMLDivElement>`, no need to add it here.

File: src/api/services/topology.ts (hunk 1 above)
Comment (mainawycliffe) [reply src/api/services/topology.ts:94]: Cool, taking care of this in a sec

File: src/components/Canary/CanaryPopup/CheckTitle.tsx (hunk 2 above)
Comment (mainawycliffe) [reply src/components/Canary/CanaryPopup/CheckTitle.tsx:9]: Yeah, that makes absolute sense.

**Comment by Prashanth-Nelli** [comment]
LGTM

**Comment by mainawycliffe** [comment]
@moshloop merging this will work on follow-up issues that may arise, and I need some of the changes for #610

**Pull request merged**